FirebaseAuthExchange/Tests/Unit/ExchangeServiceModelTests.swift

```diff
@@ -97,11 +101,41 @@ final class ModelTests: XCTestCase {
 
     let json = try encodeAsJSON(request)
 
-    XCTAssertEqual(expectedJSON, json)
+    XCTAssertEqual(json, expectedJSON)
   }
 
   private func encodeAsJSON(_ entity: Encodable) throws -> String {
     let data = try jsonEncoder.encode(entity)
     return String(data: data, encoding: .utf8)!
   }
+
+  // MARK: - Response Model Decoding Tests
+
+  func testDecodeExchangeTokenResponse() throws {
+    let accessToken = "test-access-token"
+    let timeToLive = "3.141592653s"
+    let responseJSON = """
+    {
+      "token": {
+        "accessToken": "\(accessToken)",
+        "ttl": "\(timeToLive)"
+      }
+    }
+    """
+    let expectedResponse = ExchangeTokenResponse(token: ExchangeTokenResponse.AuthExchangeToken(
+      accessToken: accessToken,
+      timeToLive: try ProtobufDuration(json: timeToLive)
+    ))
+
+    let response = try jsonDecoder.decode(ExchangeTokenResponse.self, from: responseJSON)
+
+    XCTAssertEqual(response, expectedResponse)
```

Comment:
nit: Can you also assert the expected number of seconds and nanoseconds stored in the `ProtobufDuration` object?